core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/AwsXmlUnmarshallingContext.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.protocols.xml.internal.unmarshall;
+
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.http.SdkHttpFullResponse;
+import software.amazon.awssdk.protocols.query.unmarshall.XmlElement;
+
+/**
+ * A data class to hold all the context of an unmarshalling stage for the AWS XML protocol as orchestrated by
+ * {@link AwsXmlPredicatedResponseHandler}.
+ */
+@SdkInternalApi
+public class AwsXmlUnmarshallingContext {
+    private final SdkHttpFullResponse sdkHttpFullResponse;
+    private final XmlElement parsedXml;
+    private final ExecutionAttributes executionAttributes;
+    private final Boolean responseIsSuccess;
+    private final XmlElement parsedErrorXml;
+
+    private AwsXmlUnmarshallingContext(Builder builder) {
+        this.sdkHttpFullResponse = builder.sdkHttpFullResponse;
+        this.parsedXml = builder.parsedXml;
+        this.executionAttributes = builder.executionAttributes;
+        this.responseIsSuccess = builder.responseIsSuccess;
+        this.parsedErrorXml = builder.parsedErrorXml;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * The HTTP response.
+     */
+    public SdkHttpFullResponse sdkHttpFullResponse() {
+        return sdkHttpFullResponse;
+    }
+
+    /**
+     * The parsed XML of the body, or null if there was no body.
+     */
+    public XmlElement parsedRootXml() {
+        return parsedXml;
+    }
+
+    /**
+     * The {@link ExecutionAttributes} associated with this request.
+     */
+    public ExecutionAttributes executionAttributes() {
+        return executionAttributes;
+    }
+
+    /**
+     * true if the response indicates success; false if not; null if that has not been determined yet
+     */
+    public Boolean responseIsSuccess() {
```

Comment:
`isResponseSuccess` or `isSuccessfulResponse`?

Reply:
I like isResponseSuccess more. Changing.

Reply:
Changed the name of the variable but not the method?